vision/pipeline.py

```diff
@@ -3,62 +3,115 @@
-        Filename of .bag file that the algorithm can act upon.
+    vision_communication: multiprocessing queue
+        Interface to share vision information with flight.
+    flight_communication: multiprocessing queue
+        Interface to recieve flight state information from flight.
+    camera: Camera
+        Camera to pull image from.
+    """
+    PUT_TIMEOUT = 1  # Expected time for results to be irrelevant.
 
-    env: Environment
-        The environment interface that is used by flight code.
-        The pipeline updates the interface.
+    def __init__(self, vision_communication, flight_communication, camera):
+        ##
+        self.vision_communication = vision_communication
+        self.flight_communication = flight_communication
 
-    alg_time: int
-        An integer value that corresponds to how long the video loops.
-    """
-    def __init__(self, vid_file, env, config, alg_time=98):
-        self.vid_file = vid_file
-        self.env = env
-        self.alg_time = alg_time
-        self.config = config
+        self.camera = camera.__iter__()
+
+        ##
+        prefix = 'vision' if os.path.isdir("vision") else ''
 
-    def run_algorithm(self):
+        #
+        config_filename = os.path.join(prefix, 'obstacle', 'config.json')
+
+        with open(config_filename, 'r') as config_file:
+            config = json.load(config_file)
+
+        self.obstacle_finder = ObstacleFinder(params=import_params(config))
+
+    @property
+    def picture(self):
+        return next(self.camera)
+
+    def run(self, prev_state):
         """
-        Method that takes the given video file and environment, and updates the
-        environment with detected blobs.
+        Process current camera frame.
         """
-        for i, (depth_image, color_image) in enumerate(BagFile(0, 0, 0, self.vid_file)):
-            if i == self.alg_time:
-                break
-            obstacle_finder = ObstacleFinder(params=import_params.import_params(self.config))
-            bboxes = obstacle_finder.find(color_image, depth_image)
-            self.env.update(bboxes)
+        ##
+        depth_image, color_image = self.picture
+
+        try:
+            state = self.flight_communication.get_nowait()
+        except Empty:
+            state = prev_state
 
-            #from util.obstacle_plotter import plot_obstacles
-            #plot_obstacles(obstacle_finder.keypoints, color_image)
+        ##
+        bboxes = []
+
+        if state == 'early_laps':
+            bboxes = self.obstacle_finder.find(color_image, depth_image)
+        else:
+            pass  # raise AttributeError(f"Unrecognized state: {state}")
+
+        ##
+        for bbox in bboxes:
+            self.vision_communication.put(bbox, self.PUT_TIMEOUT)
+
+        # from vision.common.blob_plotter import plot_blobs
+        # plot_blobs(self.obstacle_finder.keypoints, color_image)
+
+        return state
+
+def init_vision(vision_comm, flight_comm, video, runtime=100):
```

Comment:
```suggestion
def vision(vision_comm, flight_comm, video, runtime=100):
```
Also not sure why there is a runtime, we won't know the runtime

Reply:
And how are we going to pass you the video? We don't have it in run.py

Reply:
We dont want to name a function the same as a folder, it can be ambiguous to the interpreter.

Reply:
I told alex how to create a camera object.

Reply:
You will be enabling and disabling multiple cameras throughout, it’s gonna be visions responsibility to make those objects

Reply:
I agree it’s mildly ambiguous but it’s a one time confusion and makes more sense from the run.py perspective

Reply:
I will add that to the issue list. Its more like you cant have two things with the same name loaded ie
```python
import numpy

def numpy():
    pass

numpy.ndarray # -> error
```
So naming the function the same as the folder could limit us in the future or cause very weird bugs.

Reply:
Sure but that won’t apply here as we will only be importing this function outside of vision and you won’t ever have to import that function from inside vision

Reply:
It won't apply here but it's not good style, in my opinion.